posts/2022-10-06-memory-footprint-throughput-update.adoc

```diff
@@ -0,0 +1,101 @@
+
+_The results in the following section were all measured on a Linux X86 setup – SLES 12.3, Intel(R) Xeon(R) Platinum 8180 CPU @ 2.50GHz, 4 physical cores, 64GB RAM.  The JDK versions (and command line settings) were whatever was distributed with the default container images used for each runtime._
+
+== Memory - less memory equals cheaper infrastructure costs
+
+Cloud services make the cost and memory footprint correlation abundantly clear.  For example, the link:https://azure.microsoft.com/pricing/calculator/?service=kubernetes-service[Azure Kubernetes Service pricing] effectively doubles as you double the memory requirements of the nodes.  There's also a correlation between memory and CPUs in the pricing, so this could have a knock-on effect on your license costs.  Let's see how Open Liberty and OpenJ9 can save you memory and money.
+
+The figure below shows memory footprint comparisons (after the server has responded to the first request) for servers running the microservices benchmark link:https://github.com/blueperf[Acme Air Microservices] each using 512Mb heap.  We can see in this scenario that Open Liberty uses 1.3x – 2.9x less memory than the other runtimes we tested:
+
+[.img_border_light]
+image::img/blog/perfblog1a.png[Memory footprint of Java app servers using Acme Air Microservices application in Docker,width=70%,align="center"]
+
+If you've chosen Spring Boot for your application (yes, you can link:{url-prefix}/guides/spring-boot.html[use Spring Boot on Open Liberty]), then our measurements show an approximately 2x memory footprint benefit from link:https://developer.ibm.com/technologies/java/articles/modernize-and-optimize-spring-boot-applications/[running on Open Liberty] rather than Tomcat.  For example, the figure below shows the relative memory usage when running the link:https://github.com/spring-projects/spring-petclinic[Spring Boot Petclinic] application under load with a 4Gb heap:
+
+[.img_border_light]
+image::img/blog/perfblog2a.png[Memory footprint during load of using Spring Boot Petclinic application in Docker,width=70%,align="center"]
+
+== Throughput - higher throughput equals cheaper infrastructure and license costs
+
+The association between throughput and costs is simple: being able to put more work through a runtime means you can deploy smaller or fewer instances to satisfy demand. You’ll pay less on infrastructure and less on license costs, which are typically based on virtual CPU usage.
+
+Open Liberty also has significant throughput benefits when compared to other runtimes.  The figure below shows throughput measurements for the link:https://github.com/blueperf[Acme Air Microservices] benchmark.  We can see Open Liberty performs better than WildFly and significantly better than the other three runtimes:
```

Comment:
I think this differentiation applied to the old graph from 2020- where tomee and helidon were significantly lower. Here TomEE and JBoss are  on par with wildfly

Reply:
```suggestion
Open Liberty also has significant throughput benefits when compared to other runtimes.  The figure below shows throughput measurements for the link:https://github.com/blueperf[Acme Air Microservices] benchmark.  We can see Open Liberty performs better than WildFly and the other three runtimes:
```